src/mixins/canvas_events.mixin.js

```diff
@@ -712,6 +712,7 @@
           }
         }
       }
+      this._objectsToRender = this._chooseObjectsToRender();
```

Comment:
mmm Isn't this done before rendering? why should we do this on mouse down?

Reply:
3e091142

Reply:
Ok now i read the comment, in thise case then, shouldn't we do the same on mouse up?
there is an option to change the selection on mouseUp too rather than on mouse down.

Reply:
Now i remember we are caching the objects to render, every time we change what is selected we should also invalidate that. Right? i ll see if i spot more places.

Reply:
correct.
Mouse up might be. Isn't there some flag that makes selection run on mouseup? That's when we need to do it.